### PR TITLE
Fix staticcheck failures for pkg/scheduler/... 

### PR DESCRIPTION
**Pull request by praseodym**
**What type of PR is this?**

/kind cleanup
/priority important-longterm
/sig scheduling

**What this PR does / why we need it**:

Cleanup the `pkg/scheduler/...` issues identified by staticcheck in #81189

**Which issue(s) this PR fixes**:

#36858, #81657

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

**Review by Huang-Wei — COMMENTED** [review]
@praseodym Thanks for the PR.

I'd like all staticcheck fixes related with scheduler to be included in one PR. Could you help to fix the following packages:

- cmd/kube-scheduler
- pkg/apis/scheduling/v1alpha1
- pkg/apis/scheduling/v1beta1
- pkg/controller/volume/scheduling
- test/e2e/scheduling
- test/integration/scheduler
- test/integration/scheduler_perf
- test/integration/volumescheduling

File: pkg/scheduler/scheduler_test.go
Comment (praseodym) [reply pkg/scheduler/scheduler_test.go]: That's a lot nicer indeed, thanks! Applied and squashed into the first commit.

**Comment by praseodym** [comment]
> I'd like all staticcheck fixes related with scheduler to be included in one PR. Could you help to fix the following packages: [..]

I've created a second commit that fixes everything but `test/integration/scheduler_perf`, which has a tricky failure:
```
test/integration/scheduler_perf/scheduler_bench_test.go:359:3: should not assign to b.N (SA3001)
```
The [b.N assignment is used to enforce the minimum number of pods](https://github.com/kubernetes/kubernetes/blob/d935b1054cece13bf32e1925b5f089b475c02f3d/test/integration/scheduler_perf/scheduler_bench_test.go#L358-L360) and as far as I can see there is no proper way to achieve to set this minimum value using the regular Go benchmark functions. Would you have any suggestions for this one?

**Comment by Huang-Wei** [comment]
@praseodym Yeah, we're aware of that. You can leave it as is for now.

We will cover it in a scheduler performance improvement issue. (cc/ @draveness that staticcheck detects this issue)

**Comment by praseodym** [comment]
@Huang-Wei Alright, thanks. Any preference as to leaving that package in `.staticcheck_failures` vs. removing it and adding an ignore comment in the code itself?

**Comment by Huang-Wei** [comment]
> Any preference as to leaving that package in .staticcheck_failures vs. removing it and adding an ignore comment in the code itself?

Should be good to leave it in `.staticcheck_failures`.

**Comment by Huang-Wei** [comment]
/lgtm
/approve

@msau42 could you help /approve the changes on package `pkg/controller/volume/scheduling/`? Thanks.

@tallclair, in terms of changes in /hack, /test and pkg/apis/, do we have dedicated approver reviewing for staticcheck, or I just randomly ping the approvers? Thanks.

File: pkg/controller/volume/scheduling/scheduler_binder_test.go
Comment (msau42) [inline pkg/controller/volume/scheduling/scheduler_binder_test.go]: https://github.com/kubernetes/kubernetes/pull/81680 already covers this

Comment (Huang-Wei) [reply pkg/controller/volume/scheduling/scheduler_binder_test.go]: Got it.

@praseodym Can you revert the change in latest commit on pkg/controller/volume/scheduling (as well as the entry in .staticcheck_failures)

**Review by SataQiu — APPROVED** [review]
/approve
for hack

File: pkg/controller/volume/scheduling/scheduler_binder_test.go
Comment (praseodym) [reply pkg/controller/volume/scheduling/scheduler_binder_test.go]: Done (and rebased)!

**Comment by praseodym** [comment]
Rebased to fix conflicts.

@liggitt could you approve/review for `pkg/apis` and `test`? Thanks :slightly_smiling_face:

File: cmd/kube-scheduler/app/server.go
Comment (liggitt) [inline cmd/kube-scheduler/app/server.go]: @logicalhan does this change look right to you?

Comment (logicalhan) [reply cmd/kube-scheduler/app/server.go]: No. This is going to break our metrics. We should just add the ignore lint directive above the line for the time being.

File: pkg/scheduler/api/v1/types.go
Comment (liggitt) [inline pkg/scheduler/api/v1/types.go]: not sure what this change is about... the scheduler config API isn't proto-serialized (and "protbuf" is a typo anyway)

Comment (liggitt) [reply pkg/scheduler/api/v1/types.go]: possibly just drop the casttype directive?

Comment (praseodym) [reply pkg/scheduler/api/v1/types.go]: Good catch on the typo! Dropped the casttype directive entirely instead.

File: cmd/kube-scheduler/app/server.go
Comment (praseodym) [reply cmd/kube-scheduler/app/server.go]: Done, added as:
```go
//lint:ignore SA1019 See the Metrics Stability Migration KEP
```

**Review by logicalhan — COMMENTED** [review]
/hold cancel
/lgtm from sig-instrumentation perspective.

File: pkg/scheduler/api/v1/types.go
Comment (Huang-Wei) [reply pkg/scheduler/api/v1/types.go]: @liggitt is right, we shoudn't have `protbuf`. Interestingly, when I first reviewed the PR, I saw only "casttype=xyz" was removed.

Comment (praseodym) [reply pkg/scheduler/api/v1/types.go]: @Huang-Wei that’s because this second `casttype` instance wasn’t there in the initial version of this PR. It was introduced in https://github.com/kubernetes/kubernetes/pull/77688 which got in through a rebase, but I fixed it differently than the first `casttype` instance (my bad!).

**Pull request merged**
